### ISSUE TITLE
influence of the input parameters on estimated TE loads

### ISSUE THREAD

**Issue by KamilSJaron** [clemgoub/dnaPipeTE#26]
Hello again,

we (@reubwn, @jensbast and me) are wondering about the influence of the input parameters (genome size and sampling coverage) for the estimate of the total TE load.

The reason why we started to dig in it was that dnaPipeTE find way fewer TEs than assembly-based approaches in bdelloid rotifers (these reads: ERR2135445; genomes are published [here](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2004830)). They are difficult genomes, degenerate tetraploids and therefore it was not that clear to us, what genome size we should go for. We tested three values of input genome sizes and coverages and basically more nt we have samples, higher fraction of TEs we have annotated. You can read details in [this thread](https://github.com/KamilSJaron/genomic-features-of-asexual-animals/issues/22#issuecomment-545993111).

We hoped that it's just rotifer that is weird, but we tried the same with a genome that far more sane, a parasitoid wasp (SRR7028347), but the pattern was kind of the same. More sampled nucleotides higher TE loads detected (details latter in the same thread).

This is a bit worrying to us, as the TE load should not be dependent on the depth of sampling (at least we would hope not). We are kind of running out of inspiration how to explain the pattern. Could you help us make sense out of it?

Best,
Kamil

**Comment by clemgoub** [comment]
Hi, thanks for reaching out.
I will try to give you some ideas that may help in your case.

As you guys now, dnaPipeTE sample the genome once to assemble repeats, then performs re-samplings to increment the assembly and finally sample one last time to quantify the contings.

Because it relies on the copies similarities to assemble families, degraded copies (from old families) are harder to put together and also harder to accurately quantify. The best example for it is the human genome, where dnaPipeTE only finds half of the expected TEs. So if you expect degraded chromosomes copies this might be at play here. Did you plot a TE landscape from a repeatmasker run on an assembled genome? What is the expected age of the TE?

To set the genome size, I use the haploid value, but I have not tested dnaPipeTE with tetra or more genomes, which can add to the complexity. 

Regards to the strategy you used, it's also what I would have done. However, you might want to look what happens with much smaller values. The saturation might happen earlier, and I have never looked what happen if you push the sampling further. My experience is that if the genome is big and/or the repeat content expected is high, dnaPIpeTE is most accurate if you use small genome fractions (~0.1X). For example, with D. melanogaster (~170Mbp, ~20% TE) it fits well using ~0.25X, and for the Asian tiger mosquito (~2-3Gbp, 50% TE) I obtain saturation at 0.1X.
Usually, what I observe is that the known TE content tends to saturates, but the NA% (non annotated) is usually increasing linearly. All the values you tries (0.3-0.7) are already in the high-end of what I usually use, so maybe that's one reason.

Something typical of dnaPipeTE (for now!) is that it produces tons of contigs, most of them being at low frequency. Thus I have this arbitrary threshold in the graphs, to separate repeats that are < 0.01 % of the genome and the other ones. The increase you see is driven by these guys, that can likely be false positives, gene families, etc,... You can look what the saturation looks like separating <0.01% from >0.01% families.

So in summary, my guesses are: 1) older/degenerated copies tends to drag down the estimates. 2) too high sampling values might exhibit a weird behavior when counts are estimated (especially the < 0.01%).

Please let me know if this helped and if you need anything else,

Best,

Clém

**Comment by KamilSJaron** [comment]
Thanks a lot Clém for a quick response.

I guess it would make sense to check within out tests, what TE families are increasing in frequencies. From what you wrote, I would expect most (all?) of those are unknown.  Hmm, we need to check that out.

**Comment by clemgoub** [comment]
No problem. Yes usually in my hands the compartment increasing the fastest in saturation analyses is the "NA", while usually annotated TE tends to saturate.

My guess is the NA contains a fraction of annotated TE but probably starts to gather random genome segment as soon as the sample coverage is to large. Because by definition the non-repetitive part of the genome will no be staked into one contig (TE family), dnaPipeTE will keep assembling random parts of the genome instead of refining the discovered real TE. Thus the fraction that extends in your tests (< 0.01%) might represent this.

Let me know if you have any other question!

Cheers,

Clém

**Comment by reubwn** [comment]
Hi Clém, 

Thanks for your input, very much appreciated! 

> Usually, what I observe is that the known TE content tends to saturates, but the NA% (non annotated) is usually increasing linearly.

Yes, this is what we expected to see. But so far, we don't see any saturation even of "annotated TEs" at the higher coverage values we have used (0.3,0.5,0.7) - the plots in the other thread already exclude "NA" and "other" classes of repeats (i.e. we count only LTR, LINE, SINE, DNA, MITE + Helitron). But we will run some tests at lower coverage and see if we get any saturation, maybe `-coverage` = 0.05, 0.1, 0.15, 0.2 and 0.25, to sample the lower end of the range. Does that sound sensible? (PS I've put the `Counts.txt` data from all these runs [here](https://docs.google.com/spreadsheets/d/1dkD2yjTpkYusYVvjv_5zkAxX9n5mTnXvpeiHRCuogFs/edit?usp=sharing) if you're interested to see it.)

> Thus I have this arbitrary threshold in the graphs, to separate repeats that are < 0.01 % of the genome and the other ones. The increase you see is driven by these guys, that can likely be false positives, gene families, etc,... You can look what the saturation looks like separating <0.01% from >0.01% families.

Do you mean the difference between the dark pink and light pink in the `Bases_per_component.png` graph? and then only the repeats at <0.01% go into the `Counts.txt`, or all of them? Also, what is the relationship between the `Counts.txt` file and the RM results in the `Trinity.fasta.tbl` - are they (more or less) comparable?

Cheers,
Reuben

**Comment by clemgoub** [comment]
Hi Reuben,

I will reply in details later today. But your coverage strategy is what I would do for now too, and the Trinity.fasta.tbl is the RepeatMasker output summary for the sample reads, it is made using the Trinity.fasta.out (contigs vs Repbase or custom db). Counts.txt is inferred in part from the RepeatMasker output (Trinity.fasta.out) but the % expressed in the .tbl files represent the % of contigs covered by the Repeat annotations. The Count.txt is inferred from the # of bp of the sample reads reads mapping to the annotated contigs.

I will send you something more detailed this afternoon.

Cheers,

Clém

**Comment by mylena-s** [comment]
Sorry for my intromission, but I'm doing some genome analysis and I'm in the same situation as Reubwn. From which file could I filter the contigs to stay only with those that represent> 0.01% ?

**Comment by mylena-s** [comment]
Sorry for my intromission, but I'm doing some genome analysis and I'm in the same situation as Reuben. Im, trying to define which would be the best coverage to analyze. 
Meanwhile is there a way to filter the contigs and stay only with those that represent > 0.01%?

Mylena

**Comment by clemgoub** [comment]
Hi, sorry Mylena for the delay.

Your best shot is to filter the contigs based on the values given in the file "read_per_component_and_annotations". In this file, we report the # of bp and reads mapping to each dnaPipeTE contig. You can find the total # of bp sampled in the file "Count.txt" (total bp). 

Please let me know if you need more info!

Best,

Clément

**Comment by clemgoub** [comment]
@reubwn Do you still need extented info about your last message? -- Sorry I am in the middle of a very busy period.

Best,

Clément

**Comment by reubwn** [comment]
Hi Clém, 

No worries if you are busy at the mo, I think we are fine for the time being. I'm still interested in the discussion however - I might pick your brains a bit more at some point in the future perhaps!

Thanks anyway for your help so far, and for maintaining the software in the first place.

**Comment by clemgoub** [comment]
Sounds good then! Lmk when you need me!
